File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    const counter = {}
    
    for (const char of s) (char in counter) ? counter[char]++ : counter[char] = 1
    
    for (const char of t) {
        if (!(char in counter)) return false
        counter[char]--
    }
    
    return Object.values(counter).every((count) => count === 0)
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False

        counter = {}
        
        for i in range(len(s)):
            counter[s[i]] = counter.get(s[i], 0 ) + 1
            counter[t[i]] = counter.get(t[i], 0 ) - 1
        
        for _, v in counter.items():
            if v != 0: return False
        
        return True